### PR TITLE
Add freq and period coords to post-proc results

### LINKED ISSUE #149: core: use both omega and frequency in post-processing
`xarray` distinguishes between dimensions and coordinates. For example, you could have a dimension of `omega` that includes coordinates of radial frequency (`[rad/s]`) and `[Hz]`. For example: 

```python
import xarray as xr
import numpy as np
import matplotlib.pyplot as plt

y = np.arange(5)
da = xr.DataArray(data=y,
                  dims=['omega'],
                  coords={
                      "omega": ("omega", 2*np.pi*y, 
                                {"units":"rad/s", "long_name":"Radial frequency"}),
                      "freq": ("omega", y, 
                                {"units":"Hz", "long_name":"Frequency"})},
                  name='Data',
                  attrs={"units":" "}
                  )

fig, ax = plt.subplots(ncols=2,
                       sharey=True,
                       figsize=(6,3))

da.plot(ax=ax[0], marker='.')
da.plot.line(ax=ax[1], x='freq', marker='.')
```
image
image

### PR DESCRIPTION
**Pull request by ryancoe**
## Description
- [x] Closes #149 
- [x] Use same timestamp for both the frequency domain and time domain `Datasets`
- [x] Clean up reuse of attributes and fix a legacy unit attribute mistake

## Type of PR
- [x] New feature

## Checklist for PR
- [x] Authors read the [contribution guidelines](https://github.com/SNL-WaterPower/WecOptTool/blob/main/.github/CONTRIBUTING.md)
- [x] The pull request is from an issue branch (not main) on your fork to the [main branch in WecOptTool](https://github.com/SNL-WaterPower/WecOptTool).
- [x] The authors have given the admins edit access
- [x] All changes adhere to the style guide including PEP8, Docstrings, and Type Hints.
~~Added note in [Changelog](https://github.com/SNL-WaterPower/WecOptTool/blob/main/CHANGES.md)~~
~~Modified the documentation if applicable~~
~~Modified or added a new test~~
- [x] All tests pass
- [x] Documentation builds
- [x] [Reference or close any relevant issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

### CODE REVIEW

**Comment by ryancoe** [comment]
@cmichelenstrofer - this is ready to-go I think. Haven't checked the "Added note in [Changelog](https://github.com/SNL-WaterPower/WecOptTool/blob/main/CHANGES.md)" box, as I think you're working on automating this from the PR description/title. Only other thing to consider is whether we want to update something in the tutorials to illustrate this (e.g., similar to what I did here: https://github.com/SNL-WaterPower/WecOptTool/issues/149#issue-1424129988).

**Review by cmichelenstrofer — APPROVED** [review]
Looks great! Should we add Period as well?

**Comment by cmichelenstrofer** [comment]
Wen't through the code. Looks good! Yeah, don''t change the Change log, I'm deleting that. I like the idea of showing this is in the tutorial, up to you.

File: wecopttool/core.py [hunk 1]
```diff
@@ -793,26 +793,32 @@ def post_process(self,
...
         wave_elev_attr = {'long_name': 'Wave elevation', 'units': 'm'}
         x_wec, x_opt = self.decompose_state(res.x)
-
+        
+        freq = self.omega/2/np.pi
+        period = 1/freq
+        
+        omega_coord = ("omega", self.omega, omega_attr)
+        freq_coord = ("omega", freq, freq_attr)
+        period_coord = ("omega", period, period_attr) #TODO
```
Comment (cmichelenstrofer) [inline wecopttool/core.py:807]: #TODO can be removed

File: wecopttool/core.py [hunk 2]
```diff
@@ -793,26 +793,32 @@ def post_process(self,
...
-                                         ("omega", self.omega/2/np.pi, 
-                                          freq_attr),
-                                     'influenced_dof':
-                                         ("influenced_dof", self.dof_names,
-                                          dof_attr)},
+                                 coords={
+                                     'omega': omega_coord,
+                                     'freq': freq_coord, 
+                                     'period': period_coord,
+                                     'influenced_dof': dof_coord},
```
Comment (cmichelenstrofer) [inline wecopttool/core.py:821]: this looks much better :)

**Review by cmichelenstrofer — APPROVED** [review]

File: wecopttool/core.py (hunk 1 above)
Comment (ryancoe) [reply wecopttool/core.py]: resolved w. 0150fb3fe2c69bd929c38d2e9d02b4919f605149

**Comment by ryancoe** [comment]
Looks like [`tutorial_1_wavebot.ipynb`](https://github.com/SNL-WaterPower/WecOptTool/blob/main/examples/tutorial_1_wavebot.ipynb) doesn't actually have any plots of the frequency domain anymore. Given that its fairly long I'll not add anything for now. I'm sure an opportunity will present itself in the future.

**Pull request merged**